app/static/src/js/headscripts/gtm.js

```diff
@@ -23,6 +23,44 @@ function add_GTM() {
     GTM_Loaded = true;
 }
 
+function push_to_datalayer(event, category_code = null, category_name = null, category_traversal = null, diagnosis_result = null) {
```

Comment:
The function name is generic but the event parameters are for the diagnosed event, maybe rename and not require to feed in event name and category_transversal (global), only result/category? 

I would say not to use default values as the event shouldn't run unless we have those values. I think null is the default anyway.